src/opcodes/mod.rs

```diff
@@ -2026,36 +2030,37 @@ fn rts(cpu: &mut MOS6502, bus: &mut dyn Interface6502, address_mode_value: Addre
 // TODO: Investigate how to reuse more of the adc code
 fn sbc(cpu: &mut MOS6502, bus: &mut dyn Interface6502, address_mode_value: AddressModeValue) {
     if let AddressModeValue::AbsoluteAddress(address) = address_mode_value {
-        let value = !bus.read(address);
-        let result: u16;
+        let value = bus.read(address);
+        let carry = cpu.get_flag(StatusFlag::Carry);
 
-        // Only run if the CPU is not built in NES mode
-        // TODO: Make sure cpu is removed as dead code in nes builds
-        if cfg!(feature = "binary_coded_decimal") && cpu.get_flag(StatusFlag::Decimal) {
-            let mut sum = cpu.accumulator.wrapping_add(value).wrapping_add(cpu.get_flag(StatusFlag::Carry) as u8);
-            if (cpu.accumulator & 0x0f) + (value & 0x0f) + cpu.get_flag(StatusFlag::Carry) as u8 > 0x09 {
-                sum = sum.wrapping_sub(0x06);
-            }
-            if (sum & 0xf0) > 0x90 {
-                sum = sum.wrapping_sub(0x60);
-                cpu.set_flag(StatusFlag::Carry, false);
-            } else {
-                cpu.set_flag(StatusFlag::Carry, true);
-            }
-            result = u16::from(sum);
-        } else {
-            result = u16::from(cpu.accumulator) + u16::from(value) + cpu.get_flag(StatusFlag::Carry) as u16;
-            // Set the Carry flag for chain adding multi byte numbers
-            cpu.set_flag(StatusFlag::Carry, result > u16::from(u8::max_value()));
-        }
+        let mut result = u16::from(cpu.accumulator) + u16::from(!value) + carry as u16;
+        // Set the Carry flag for chain adding multi byte numbers
+        cpu.set_flag(StatusFlag::Carry, result > u16::from(u8::max_value()));
         cpu.set_flag(StatusFlag::Zero, result as u8 == 0);
         // Set the Overflow flag if a signed overflow has occurred
         cpu.set_flag(
             StatusFlag::Overflow,
-            (!(cpu.accumulator ^ value) & (cpu.accumulator ^ result as u8) & StatusFlag::Negative as u8) > 0,
+            ((cpu.accumulator ^ value) & (cpu.accumulator ^ result as u8) & StatusFlag::Negative as u8) > 0,
         );
         // Negative flag is in bit 7, so it can be used to test if the result is negative, because a negative value will also have a 1 in bit 7
         cpu.set_flag(StatusFlag::Negative, result as u8 & StatusFlag::Negative as u8 > 0);
+
+        // Only run if the CPU is not built in NES mode
+        // TODO: Make sure cpu is removed as dead code in nes builds
+        if cfg!(feature = "binary_coded_decimal") && cpu.get_flag(StatusFlag::Decimal) {
+            let value = value as i16;
+
+            let mut al= (cpu.accumulator & 0xf) as i16 - (value & 0xf) + carry as i16 - 1;
```

Comment:
```suggestion
            let mut al = (cpu.accumulator & 0xf) as i16 - (value & 0xf) + carry as i16 - 1;
```

Just a small style thing. I'd also like `al` to be renamed to something more specific here.